### PR TITLE
fix: configure remote grpc file transfer service

### PR DESCRIPTION
**Pull request by hpohekar**
closes https://github.com/ansys/pyfluent/issues/2782

Set `pyfluent.USE_FILE_TRANSFER_SERVICE=True` to use file transfer service.

### CODE REVIEW

File: src/ansys/fluent/core/__init__.py [hunk 1]
```diff
@@ -84,6 +84,9 @@ def version_info() -> str:
 # Whether stream and cache commands state
 DATAMODEL_USE_NOCOMMANDS_DIFF_STATE = True
 
+# Whether to use remote gRPC file transfer service
+REMOTE_GRPC_FILE_TRANSFER_SERVICE = None
```
Comment (mkundu1) [inline src/ansys/fluent/core/__init__.py:88]: It is simpler if a boolean config is either True or False.

**Comment by mkundu1** [comment]
I think we should keep the source code conditional based on the configuration without changing the tests.

**Comment by hpohekar** [comment]
> I think we should keep the source code conditional based on the configuration without changing the tests. After fixing the underlying issue, we shall be able to run the same tests after switching the configuration.

@mkundu1 

The issue is that if we set `pyfluent.REMOTE_GRPC_FILE_TRANSFER_SERVICE=True` then 

`examples.download_file("Static_Mixer_Parameters.cas.h5", "pyfluent/static_mixer",)` returns `"Static_Mixer_Parameters.cas.h5"` only, which is incorrect because we are changing `host_mount_path` to `pyfluent.USER_DATA_PATH` to avoid file modification issue.

We need full path therefore we are updating the code as follows

```
if pyfluent.REMOTE_GRPC_FILE_TRANSFER_SERVICE:
    file_name = examples.download_file("Static_Mixer_Parameters.cas.h5", "pyfluent/static_mixer",, return_without_path=False)
else:
    file_name = examples.download_file("Static_Mixer_Parameters.cas.h5", "pyfluent/static_mixer",)
```

**Comment by hpohekar** [comment]
> I think we should keep the source code conditional based on the configuration without changing the tests. After fixing the underlying issue, we shall be able to run the same tests after switching the configuration.

@mkundu1 

In following PR also we are changing `host_mount_path` and updating tests accordingly. (we are holding it for now due to hang issue)

https://github.com/ansys/pyfluent/pull/2743

**Comment by hpohekar** [comment]
Ideally all tests should use full path but we had done this adjustment because we have set `pyfluent.EXAMPLES_PATH` as a `host_mount_path` before using remote gRPC FTS.

**Comment by mkundu1** [comment]
> > I think we should keep the source code conditional based on the configuration without changing the tests. After fixing the underlying issue, we shall be able to run the same tests after switching the configuration.
> 
> @mkundu1
> 
> In following PR also we are changing `host_mount_path` and updating tests accordingly. (we are holding it for now due to hang issue)
> 
> #2743

Does it

File: src/ansys/fluent/core/__init__.py (hunk 1 above)
Comment (hpohekar) [reply src/ansys/fluent/core/__init__.py]: Done. Thank you.

**Review by prmukherj — APPROVED** [review]

**Comment by seanpearsonuk** [comment]
> I think we should keep the source code conditional based on the configuration without changing the tests. After fixing the underlying issue, we shall be able to run the same tests after switching the configuration.

Yes, I was going to make the same comment, thanks @mkundu1. It's a simple case of encapsulation.

And, @hpohekar, irrespective of this point, please look at the code you have written in all these tests, simplified as follows. Can you see the issue?
```
if x:
    y = foo(a, b, False)
else:
    y = foo(a, b, True)
```

**Review by seanpearsonuk — APPROVED** [review]

File: src/ansys/fluent/core/__init__.py (hunk 1 above)
Comment (hpohekar) [reply src/ansys/fluent/core/__init__.py]: Done. Thank you.

**Comment by hpohekar** [comment]
@mkundu1 @seanpearsonuk 

Closing due to consistent permission errors.

We have disabled remote gRPC FTS in [this PR.](https://github.com/ansys/pyfluent/pull/2780). 

Nightly Dev Test is not hanging now. https://github.com/ansys/pyfluent/actions/runs/8949142087/job/24583086511

**Comment by hpohekar** [comment]
https://github.com/ansys/pyfluent/pull/2783

**Pull request merged**